lib/Sema/CSSimplify.cpp

```diff
@@ -2972,6 +2972,11 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       
     } else {
       if (!hasStaticMembers) {
+        if (cand->isImplicit() && isa<ConstructorDecl>(cand)) {
```

Comment:
When we synthesize the inherited initializer, we can check if the constructor it will delegate to is already invalid, and if so, skip synthesizing. See Sema/CodeSynthesis.cpp, swift::createDesignatedInitOverride(). This method is allowed to return nullptr to mean "I refuse to synthesize this constructor".